src/com/xilinx/rapidwright/rwroute/PartialRouter.java

```diff
@@ -218,47 +243,82 @@ protected void routeGlobalClkNets() {
-                        if (preservedNet == clk) {
-                            continue;
-                        }
-                        if (preservedNet == null) {
-                            // Assume this node has already been unpreserved
-                        } else {
-                            unpreserveNet(preservedNet);
-                            unpreserveNets.add(preservedNet);
-                        }
-
-                        // Redo preserving clk
-                        Net oldNet = routingGraph.preserve(node, clk);
-                        assert(oldNet == null);
-
-                        // Clear preservedNode's prev pointer so that it doesn't get misinterpreted
+        List<Net> unpreserveNets = unpreserveCongestedNets(clkNets);
+        if (!unpreserveNets.isEmpty()) {
+            System.out.println("INFO: Unpreserving " + unpreserveNets.size() + " nets due to clock congestion");
+            for (Net net : unpreserveNets) {
+                System.out.println("\t" + net);
+            }
+        }
+    }
+
+    @Override
+    protected void routeStaticNets() {
+        if (staticNetAndRoutingTargets.isEmpty())
+            return;
+
+        super.routeStaticNets();
+
+        List<Net> unpreserveNets = unpreserveCongestedNets(staticNetAndRoutingTargets.keySet());
+        if (!unpreserveNets.isEmpty()) {
+            System.out.println("INFO: Unpreserving " + unpreserveNets.size() + " nets due to static net congestion");
+            for (Net net : unpreserveNets) {
+                System.out.println("\t" + net);
+            }
+        }
+    }
+
+    private List<Net> unpreserveCongestedNets(Collection<Net> globalNets) {
+        if (!softPreserve) {
+            return Collections.emptyList();
+        }
+
+        // Even though route{GlobalClk,Static}Nets() has called preserveNet() for all its nets,
+        // it will not overwrite those nodes which have already been preserved by other nets.
+        // Discover such occurrences so that the entire 'victim' net can be correctly
+        // unpreserved (thus re-routed) and re-preserve the global.
+        List<Net> unpreserveNets = new ArrayList<>();
+        for (Net net : globalNets) {
+            for (PIP pip : net.getPIPs()) {
+                for (Node node : Arrays.asList(pip.getStartNode(), pip.getEndNode())) {
+                    Net preservedNet = routingGraph.getPreservedNet(node);
+                    if (preservedNet == net) {
+                        continue;
+                    }
+                    if (preservedNet == null) {
+                        // Assume this node has already been unpreserved
+                    } else if (RouterHelper.isRoutableNetWithSourceSinks(preservedNet)) {
+                        unpreserveNet(preservedNet);
+                        unpreserveNets.add(preservedNet);
+                    }
+
+                    // Redo preserving clk
```

Comment:
This comment might need to be reworded based on the change from `clkNets -> globalNets`.